### PR TITLE
Adding release note generator action based on k8s release notes generator tool

### PR DESCRIPTION
**Pull request by n3wscott**
Fixes https://github.com/cloudevents/spec/issues/817

Signed-off-by: Scott Nichols <snichols@vmware.com>

### CODE REVIEW

File: .github/workflows/release-notes.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            --required-author "" \
+            --repo-path       "$(pwd)" \
+            --output          release-notes.md
+
+      - name: Display Notes
+        run: |
+          cat release-notes.md
+
+      - name: Archive Release Notes
+        uses: actions/upload-artifact@v2
```
Comment (duglin) [inline .github/workflows/release-notes.yaml:88]: Just want to make sure I followed the docs for this action correctly... the artifact will be uploaded to a special spot in the workflow and then at release time we can just copy-n-paste the results to where ever we want, right?

File: .github/workflows/release-notes.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'Release Notes'
+
+on:
```
[the remark this answers is not in the document]
Comment (n3wscott) [reply .github/workflows/release-notes.yaml:18]: it is only manually run

File: .github/workflows/release-notes.yaml (hunk 1 above)
Comment (n3wscott) [reply .github/workflows/release-notes.yaml:88]: it attaches the release note to the workflow run, see an example: https://github.com/knative/eventing/actions/runs/839442538

**Comment by duglin** [comment]
Approved on the 5/27 call

**Pull request merged**